skbio/metadata/_testing.py

```diff
@@ -904,3 +905,245 @@ def test_has_positional_metadata(self):
+        self.assertReallyEqual(obj1, obj2)
+
+    def test_ne_diff_len(self):
+        obj1 = self._interval_metadata_constructor_(0)
+        obj2 = self._interval_metadata_constructor_(self.upper_bound)
+        self.assertReallyNotEqual(obj1, obj2)
+
+    def test_ne_only_one_is_empty(self):
+        im1 = IntervalMetadata(self.upper_bound)
+        im1.add(**self.intvls[0])
+        obj1 = self._interval_metadata_constructor_(self.upper_bound, im1)
+
+        obj2 = self._interval_metadata_constructor_(self.upper_bound)
+
+        self.assertReallyNotEqual(obj1, obj2)
+
+    def test_ne(self):
+        im1 = IntervalMetadata(self.upper_bound)
+        im1.add(**self.intvls[0])
+        obj1 = self._interval_metadata_constructor_(self.upper_bound, im1)
+
+        im2 = IntervalMetadata(self.upper_bound)
+        im2.add(**self.intvls[1])
+        obj2 = self._interval_metadata_constructor_(self.upper_bound, im2)
+
+        self.assertReallyNotEqual(obj1, obj2)
+
+    def test_copy_interval_metadata_empty(self):
+        obj = self._interval_metadata_constructor_(self.upper_bound, self.im)
+        obj_copy = copy.copy(obj)
+
+        self.assertEqual(obj, obj_copy)
+        self.assertIsNot(obj, obj_copy)
+
+    def test_copy_interval_metadata(self):
+        self.im.add(**self.intvls[0])
+        obj = self._interval_metadata_constructor_(self.upper_bound, self.im)
+        obj_copy = copy.copy(obj)
+
+        self.assertEqual(obj, obj_copy)
+        self.assertIsNot(obj, obj_copy)
+
+        self.assertIsNot(obj.interval_metadata,
+                         obj_copy.interval_metadata)
+        self.assertIsNot(obj.interval_metadata._intervals,
+                         obj_copy.interval_metadata._intervals)
+        for i, j in zip(obj.interval_metadata._intervals,
+                        obj_copy.interval_metadata._intervals):
+            self.assertIsNot(i, j)
+            self.assertIsNot(i.metadata, j.metadata)
+            for k in i.metadata:
+                self.assertIs(i.metadata[k], j.metadata[k])
+
+    def test_deepcopy_interval_metadata(self):
+        self.im.add(**self.intvls[1])
+        obj = self._interval_metadata_constructor_(self.upper_bound, self.im)
+        obj_copy = copy.deepcopy(obj)
+
+        self.assertEqual(obj, obj_copy)
+        self.assertIsNot(obj, obj_copy)
+
+        self.assertIsNot(obj.interval_metadata,
+                         obj_copy.interval_metadata)
+        self.assertIsNot(obj.interval_metadata._intervals,
+                         obj_copy.interval_metadata._intervals)
+        for i, j in zip(obj.interval_metadata._intervals,
+                        obj_copy.interval_metadata._intervals):
+            self.assertIsNot(i, j)
+            self.assertIsNot(i.metadata, j.metadata)
+            self.assertIsNot(i.metadata['gene'], j.metadata['gene'])
+            self.assertIs(i.metadata['product'], j.metadata['product'])
+
+    def test_deepcopy_interval_metadata_empty(self):
```

Comment:
add more assertions similar to those in `test_deepcopy_positional_metadata_empty`